Ebusd integration

CrazYoshi opened the pull request:
## Description:
Integration between ebusd (https://github.com/john30/ebusd), daemon for communication with eBUS heating systems, and homeassistant using sensor component.

## Example entry for `configuration.yaml`:
```
ebus:
  - platform: ebusd
    host: 127.0.0.1
    port: 8888
    name: 'ebusd'
    circuit: '700'
    monitored_conditions:
      - 'MaxFlowTemperatureDesired'
      - 'MinFlowTemperatureDesired'
      - 'WaterPressure'
      - 'PumpStatus'
      - 'HWTemperatureDesired'
      - 'HWTimerMonday'
      - 'HWTimerTuesday'
      - 'HWTimerWednesday'
      - 'HWTimerThursday'
      - 'HWTimerFriday'
      - 'HWTimerSaturday'
      - 'HWTimerSunday'
      - 'Zone1NightTemperature'
      - 'Zone1DayTemperature'
      - 'Zone1RoomTemperature'
      - 'Zone1ActualRoomTemperatureDesired'
      - 'Zone1TimerMonday'
      - 'Zone1TimerTuesday'
      - 'Zone1TimerWednesday'
      - 'Zone1TimerThursday'
      - 'Zone1TimerFriday'
      - 'Zone1TimerSaturday'
      - 'Zone1TimerSunday'
      - 'Zone1OperativeMode'
      - 'PowerEnergyConsumptionThisMonth'
```

CrazYoshi committed:
Ebusd integration

Integration between ebusd (https://github.com/john30/ebusd), daemon for communication with eBUS heating systems, and homeassistant using sensor component.

CrazYoshi committed:
requirements_all updated

update requirements_all

CrazYoshi committed:
New-style class for EbusdData

Solve: "Class 'EbusdData' inherits from object, can be safely removed from bases"

CrazYoshi committed:
hund string length fixes

hund string length fixes

cgarwood commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,251 @@
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=[]):
```
Please switch CONF_MONITORED_VARIABLES throughout the code to CONF_MONITORED_CONDITIONS - it was just recently decided to deprecate CONF_MONITORED_VARIABLES in favor of CONF_MONITORED_CONDITIONS.

cgarwood reviewed (changes requested):
Tests will need added for the component as well.

CrazYoshi committed:
CONF_MONITORED_VARIABLES to CONF_MONITORED_CONDITIONS

CONF_MONITORED_VARIABLES changed to CONF_MONITORED_CONDITIONS

CrazYoshi committed:
coveragerc updated

coveragerc updated

frenck commented:
Could not find a related documentation PR, therefore added the label `docs-missing`.

CrazYoshi commented:
> Could not find a related documentation PR, therefore added the label `docs-missing`.

Pull request submitted to add documentation to home-assistant.io

balloob commented on .coveragerc:
```diff
@@ -662,6 +662,7 @@ omit =
     homeassistant/components/sensor/duke_energy.py
     homeassistant/components/sensor/dwd_weather_warnings.py
     homeassistant/components/sensor/ebox.py
+	homeassistant/components/sensor/ebusd.py
```
Please use spaces instead of tab

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,251 @@
+        add_devices(dev)
+        hass.services.register('sensor', 'ebusd_write', data.write)
+    except socket.timeout:
+        _LOGGER.error("socket timeout error")
```
Raise `PlatformNotReady`

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,251 @@
+                self.value[name] = command_result
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+            raise RuntimeError("socket timeout")
```
Don't raise errors, as that will log exceptions. You can set `available` property to False.

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,251 @@
+                elif self._type == 3:
+                    self._state = self.data.value[self._name]
+        except RuntimeError:
+            _LOGGER.debug("EbusdData.update exception")
```
Set `available` property to False when this happens.

CrazYoshi replied:
That will be an issue cause most of the time ebusd returns an error so the sensor value will blink between not available and the correct value. In my opinion it is better to have the previous stored value instead of nothing.

Dinth replied:
I agree with CrazYoshi, Ebus is a serial interface with signal carried usually over two straight (not screened, not twisted) cables.  Theoretically random interferences can appear, but shouldn't disrupt the functionality.

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,251 @@
+        """Fetch new state data for the sensor."""
+        try:
+            self.data.update(self._name)
+            if self._name in self.data.value:
```
Use guard clause:

```python
if self.name not in self.data.value:
    return
```

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,251 @@
+        return
+
+
+def timer_format(string):
```
We should prefer storing times in isoformat.

CrazYoshi replied:
what do you mean with ISO format? Is it only HH:mm not a datetime

Dinth commented:
> i tried to add your component as a custom component and the configuration seems ok so far (at least the check says so). I followed your description, but i do not see any values in the dashboard. How do i add them?
Is your ebus daemon running and are you sure that you're fetching variables which are supported by your hardware?

dkuschmierz commented:
> > i tried to add your component as a custom component and the configuration seems ok so far (at least the check says so). I followed your description, but i do not see any values in the dashboard. How do i add them?
> 
> Is your ebus daemon running and are you sure that you're fetching variables which are supported by your hardware?

Yes, my ebus daemon is running and i can read the desired value. Unfortunately it was not in the config, so I added the following line:

`'HwcTemperature':
	['HwcTemp', '°C', 'mdi:thermometer', 0],`

dkuschmierz commented:
And the config looks like that:

# Weather prediction
sensor:
  - platform: yr

#ebusd
  - platform: ebusd
    host: 127.0.0.1
    port: 8888
    name: 'ebusd'
    circuit: 'ehp'
    monitored_variables:
      - 'HwcTemperature'

CrazYoshi committed:
switched back to CONF_MONITORED_VARIABLES

CONF_MONITORED_VARIABLES restored, raise PlatformNotReady exception and guard clause added on update()

CrazYoshi committed:
Added HWTemperature to configurable variables

Added HWTemperature to configurable variables

CrazYoshi commented:
> And the config looks like that:
> 
> # Weather prediction
> sensor:
> 
> * platform: yr
> 
> #ebusd
> 
> * platform: ebusd
>   host: 127.0.0.1
>   port: 8888
>   name: 'ebusd'
>   circuit: 'ehp'
>   monitored_variables:
>   
>   * 'HwcTemperature'

I have added HWTemperature to configurable variables. I have also changed back to CONF_MONITORED_VARIABLES beacause I have tested my component with CONF_MONITORED_CONDITIONS and it doens't work anymore

CrazYoshi committed:
revert to CONF_MONITORED_CONDITIONS

revert to CONF_MONITORED_CONDITIONS

dkuschmierz commented:
Hi, i downloaded the latest ebusd.py with your latest changes, but it's still not working...

CrazYoshi commented:
> Hi, i downloaded the latest ebusd.py with your latest changes, but it's still not working...

Do you have changed your configuration using monitored_conditions instead of monitored_variables in your configuration.yaml?

CrazYoshi commented:
> Hi, indeed that was the problem. But i found a problem with the decimal parsing as my heat pump give me the temperature in a way like 56.6;ok. It's the temperature with a status.
> 
> Can you please modify the decimal parsing like that:
> 
> 1. add `import re`
> 2. Modify the decimal parsing as follows:
>    `if self._type == 0: n = re.search(r"[-+]?\d*\.\d+|\d+",self.data.value[self._name]) self._state = format(float(n.group(0)), '.1f')`
> 
> This way it will take the first float it recognises. It tried it for myself and it's working pretty well.
> 
> Another question. Are the values only polled if I have a opened dashboard or is it polling the data even if nobody is watching the dashboard?

This is seems an issue in ebusd itself that returns the wrong value in my opinion. Maybe you should open an issue to ebusd or ebusd configuration?

dkuschmierz commented:
Ok, the values are updated in background. Setting the CACHE_TTL to 120 and the values are now updated every 120s

dkuschmierz commented:
The value from ebusd, with ;ok is. correct. The Message from my Vaillant heatpump consists of the Temperature and the Status of the measurement

CrazYoshi commented:
> Ok, the values are updated in background. Setting the CACHE_TTL to 120 and the values are now updated every 120s

The values are updated in background without setting any variables in my case. In my opinion the value returned should be temperature only like the other decimal values that I get from the deamon

dkuschmierz commented:
Yes it does, but if you set the cache to 900 ebusctl does not query the data from the bus if the last query was within 15 minutes. It returns the last query then. 

I think ebusctl is working correct here, as it only returns the data which are comming over the bus! As the bus telegramm consists of the temperature and the status, it‘s correct to return both. Maybe somebody is interested at the status as well.

> Am 03.10.2018 um 22:41 schrieb CrazYoshi <notifications@github.com>:
> 
> Ok, the values are updated in background. Setting the CACHE_TTL to 120 and the values are now updated every 120s
> 
> The values are updated in background without setting any variables in my case. In my opinion the value returned should be temperature only like the other decimal values that I get from the deamon
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

CrazYoshi commented:
If ebusd must return the value temperature;status I will change the code in order to return the value without any conversion so you will receive 56.6;ok and you can take the temperature using a template component. I think it is the best solution

CrazYoshi committed:
ebusd opmode translation

ebusd opmode translation

CrazYoshi committed:
remove comment over 79 char

remove comment over 79 char

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,254 @@
+    def update(self, name):
+        """Call the Ebusd API to update the data."""
+        import ebusdpy
+        command = READ_COMMAND.format(self._circuit, name, CACHE_TTL)
```
This should be in ebusdpy.

CrazYoshi replied:
Doesn't male sense to have it in ebusdpy library. It should contain only the methods to read/write from socket

balloob commented on homeassistant/components/sensor/ebusd.py:
```diff
@@ -0,0 +1,254 @@
+            _LOGGER.error("socket error: %s", socket.error)
+            raise RuntimeError("Command failed")
+
+    def write(self, call):
```
Sensors don't write. If writing is a possibility, this should be a component ?

balloob commented on homeassistant/components/sensor/strings.ebusd.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "state": {
+    "day": "Day",
```
This is not a sensor but a binary sensor (two states). Also, I don't see a valid use case that a user would use ebusd to get if it's currently day or night. So let's remove it.

CrazYoshi replied:
Operative mode have four values: on, off, day, night. It also update automatically like most sensors so you will see the change in history and create automations. So I will not remove it

CrazYoshi committed:
changed from sensor to component

changed from sensor to component

CrazYoshi committed:
hund fixes

hund fixes

CrazYoshi committed:
hound fixes

hound fixes

CrazYoshi committed:
coveragerc fix

coveragerc fix

CrazYoshi committed:
requirements_all fix

requirements_all fix

dkuschmierz commented:
Hi, 

can you please give an example how to show the values with the new components settings?

CrazYoshi commented:
> Hi,
> 
> can you please give an example how to show the values with the new components settings?

You can find the configuration.yaml example at the beginning of this pull request. Values will be shown in states (eg: ebus.ebusd_hwctempdesired)

dkuschmierz commented:
Ah, i see a problem in the logs:

```
[custom_components.ebus] Error while setting up platform ebusd
  File "/home/homeassistant/.homeassistant/custom_components/ebus/ebusd.py", line 64, in setup_platform
  File "/home/homeassistant/.homeassistant/custom_components/ebus/ebusd.py", line 139, in __init__
```

CrazYoshi commented:
> Ah, i see a problem in the logs:
> 
> ```
> [custom_components.ebus] Error while setting up platform ebusd
>   File "/home/homeassistant/.homeassistant/custom_components/ebus/ebusd.py", line 64, in setup_platform
>   File "/home/homeassistant/.homeassistant/custom_components/ebus/ebusd.py", line 139, in __init__
> ```

I have tested on my environment and I have no error so far. Are you sure to have downloaded also consts.py?
I will try again later

dkuschmierz commented:
Yes sure, i cloned the tag and copied the whole folder...

CrazYoshi commented:
> Yes sure, i cloned the tag and copied the whole folder...

Could you please send your configuration?

dkuschmierz commented:
ebus:
  - platform: ebusd
    host: 127.0.0.1
    port: 8888
    name: 'ebusd'
    circuit: 'ehp'
    monitored_conditions:
      - 'HWTemperature'
      - 'OutsideTemp'

dkuschmierz commented:
Ok, as i Posted my config i found the Problem. OutsideTemp was a measurement value i added to your script

Dinth commented:
> If ebusd must return the value temperature;status I will change the code in order to return the value without any conversion so you will receive 56.6;ok and you can take the temperature using a template component. I think it is the best solution

Hi. Could you please do the same with WaterPressure sensor?

`ValueError: could not convert string to float: '1.523;ok'`

dkuschmierz commented:
Can you please add the new type in the update function in Ebusd.py:
`elif self._type == 4:
                self._valTemp, self._stateTemp = self.data.value[self._name].split(";")
                if "ok" in self._stateTemp:
                    self._state = format(float(self._valTemp),'.1f')`

Can you please add the following value in the const.py:
`    'OutsideTemp':
        ['OutsideTemp', 'C', 'mdi:thermometer',4]`

And would it be possible to give each sensor an individual update time? For example it would be sufficient to read the outside temperature every 5 minutes, but some pump states, for example circulation pump, should be read every 10s to get the switching cycles...

Dinth commented:
> Can you please add the new type in the update function in Ebusd.py:

I believe it would be better to have this merged to the main branch asap and once it's merged we can focus on adding more variables and i will be happy to commit some myself. There's no point of adding new variables now, as every boiler has an unique set and all together there are probably thousands of different variables.
`ebusctl find | wc -l
172`

Dinth commented:
Another sughestion, I believe that the platform name should be changed to something like ebus_sensor, just to avoid breaking change when adding switches, etc in the future

CrazYoshi commented:
> > If ebusd must return the value temperature;status I will change the code in order to return the value without any conversion so you will receive 56.6;ok and you can take the temperature using a template component. I think it is the best solution
> 
> Hi. Could you please do the same with WaterPressure sensor?
> 
> `ValueError: could not convert string to float: '1.523;ok'`
> 
> Other than that i tested the component and it seems to work great

No sorry I can't do that. In my ebusd I receive it without ';ok' at the end. I keep thinking that is an issue on ebusd configuration that give you the status together with value..

CrazYoshi committed:
const management for different circuit

const management for different circuit

CrazYoshi committed:
hound fixes

hound fixes

CrazYoshi committed:
trailing whitespace

trailing whitespace

CrazYoshi commented:
> > 'OutsideTemp': ['OutsideTemp', 'C', 'mdi:thermometer',4]
> 
> I believe it would be better to have this merged to the main branch asap and once it's merged we can focus on adding more variables/features and i will be happy to commit some myself. There's no point of adding new variables now, as every device has an unique set and all together there are probably thousands of different variables out there
> `ebusctl find -r | wc -l`
> `163`
> 
> > elif self._type == 4: self._valTemp, self._stateTemp = self.data.value[self._name].split(";") if "ok" in self._stateTemp: self._state = format(float(self._valTemp),'.1f')
> 
> Same here, this should be added in the future as "$temp; ok" value is correct - that's what the boiler returns, status can be split using template sensor and last but not least - formatting return values to look nice on every device will be very time consuming.

I have now change the code in order to have in const.py sensor types splitted by circuit and I also create the type 4 for value;status return values

CrazYoshi committed:
Access to a protected member fix

Access to a protected member fix

CrazYoshi commented:
Is it everthing fine with my component? When it will be merged on master?

dkuschmierz commented:
Can you please explain how the different values are updated? At the moment i have approx. 20 values I read from my heatpump, but it seems that the values are read in serial and not in parallel. Is this correct? My wish is, that I can read values in parallel with different query cycles. For example, i will read the outside-temperature every 2 minutes, but the circulation pump every 5s to check if it is running. Is this anyhow possible?

CrazYoshi commented:
> Can you please explain how the different values are updated? At the moment i have approx. 20 values I read from my heatpump, but it seems that the values are read in serial and not in parallel. Is this correct? My wish is, that I can read values in parallel with different query cycles. For example, i will read the outside-temperature every 2 minutes, but the circulation pump every 5s to check if it is running. Is this anyhow possible?

No, this is not possible cause if you already open a socket connection if you open another one in parallel it will receive an error cause socket cannot manage more than one connection open at time.

dkuschmierz commented:
> > Can you please explain how the different values are updated? At the moment i have approx. 20 values I read from my heatpump, but it seems that the values are read in serial and not in parallel. Is this correct? My wish is, that I can read values in parallel with different query cycles. For example, i will read the outside-temperature every 2 minutes, but the circulation pump every 5s to check if it is running. Is this anyhow possible?
> 
> No, this is not possible cause if you already open a socket connection if you open another one in parallel it will receive an error cause socket cannot manage more than one connection open at time.

But what is exactly the MINIMUM_TIME_BETWEEN_UPDATE parameter? Is this the time which is waited between Sensor_A_Value and Sensor_B_Value or is this the time between Sensor_A_Value and Sensor_A_Value2? And if it is the time between to different sensor values, are they read in serial or random?

Dinth commented:
> Ok, but then i would expect that i will see **all** sensor-values after this time if i restart the system. But for some values it takes up to 5 minutes until I see the first value...

It may depend on how often your boiler broadcasts the value. AFAIK with eBus protocol devices (boilers, etc) periodically broadcast variables on their own. Maybe it's possible to query the device (im not sure about this) specifically for a particular variable, but during normal usage the boiler just periodically broadcasts values and eBusd picks them up. 
Can you ssh to your eBusd server and try running ebusctl read to check if the value has been broadcasted by the boiler?

dkuschmierz commented:
I think that it’s more a home-assistant problem. The value I’m observing is not a brodcasted, it’s a queried one. Even if I query the value with ebusctl the value is not updated in homeassistant..

> Am 16.10.2018 um 16:38 schrieb Michal <notifications@github.com>:
> 
> Ok, but then i would expect that i will see all sensor-values after this time if i restart the system. But for some values it takes up to 5 minutes until I see the first value...
> 
> It may depend on how often your boiler broadcasts the value. AFAIK with eBus protocol devices (boilers, etc) periodically broadcast variables on their own. Maybe it's possible to query the device (im not sure about this) specifically for a particular variable, but during normal usage the boiler just periodically broadcasts values and eBusd picks them up.
> Can you ssh to your eBusd server and try running ebusctl read to check if the value has been broadcasted by the boiler?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

CrazYoshi commented:
> I think that it’s more a home-assistant problem. The value I’m observing is not a brodcasted, it’s a queried one. Even if I query the value with ebusctl the value is not updated in homeassistant..
> […](#)
> Am 16.10.2018 um 16:38 schrieb Michal ***@***.***>: Ok, but then i would expect that i will see all sensor-values after this time if i restart the system. But for some values it takes up to 5 minutes until I see the first value... It may depend on how often your boiler broadcasts the value. AFAIK with eBus protocol devices (boilers, etc) periodically broadcast variables on their own. Maybe it's possible to query the device (im not sure about this) specifically for a particular variable, but during normal usage the boiler just periodically broadcasts values and eBusd picks them up. Can you ssh to your eBusd server and try running ebusctl read to check if the value has been broadcasted by the boiler? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

So I think is not an home-assistant issue too cause ebusctl open a socket connection as my component do

Dinth commented:
> I think that it’s more a home-assistant problem. The value I’m observing is not a brodcasted, it’s a queried one. Even if I query the value with ebusctl the value is not updated in homeassistant..

It's really hard to tell without further investigation. This sensor doesnt speak with the device directly, but through an external daemon/server. 
Boiler <-> ebusd "receiver" <-> ebusd server <-> home assistant component
The best wait to check where's the problem would be to query a value simulatenously on both ebusd server and Home Assistant. 

> The value I’m observing is not a brodcasted, it’s a queried one.
Correct me if im wrong - but even if manually querying values is possible you could only do that by manually running a relevant ebusctl command, otherwise your rely on broadcasted values.

dkuschmierz commented:
Ok, but what happens if two components try to connect at the same time? Is one blocked? And if yes, how often does it try to reconnect?

> Am 16.10.2018 um 16:59 schrieb CrazYoshi <notifications@github.com>:
> 
> I think that it’s more a home-assistant problem. The value I’m observing is not a brodcasted, it’s a queried one. Even if I query the value with ebusctl the value is not updated in homeassistant..
> …
> Am 16.10.2018 um 16:38 schrieb Michal ***@***.***>: Ok, but then i would expect that i will see all sensor-values after this time if i restart the system. But for some values it takes up to 5 minutes until I see the first value... It may depend on how often your boiler broadcasts the value. AFAIK with eBus protocol devices (boilers, etc) periodically broadcast variables on their own. Maybe it's possible to query the device (im not sure about this) specifically for a particular variable, but during normal usage the boiler just periodically broadcasts values and eBusd picks them up. Can you ssh to your eBusd server and try running ebusctl read to check if the value has been broadcasted by the boiler? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.
> 
> So I think is not an home-assistant issue too cause ebusctl open a socket connection as my component do
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

CrazYoshi commented:
> Ok, but what happens if two components try to connect at the same time? Is one blocked? And if yes, how often does it try to reconnect?
> […](#)
> Am 16.10.2018 um 16:59 schrieb CrazYoshi ***@***.***>: I think that it’s more a home-assistant problem. The value I’m observing is not a brodcasted, it’s a queried one. Even if I query the value with ebusctl the value is not updated in homeassistant.. … Am 16.10.2018 um 16:38 schrieb Michal ***@***.***>: Ok, but then i would expect that i will see all sensor-values after this time if i restart the system. But for some values it takes up to 5 minutes until I see the first value... It may depend on how often your boiler broadcasts the value. AFAIK with eBus protocol devices (boilers, etc) periodically broadcast variables on their own. Maybe it's possible to query the device (im not sure about this) specifically for a particular variable, but during normal usage the boiler just periodically broadcasts values and eBusd picks them up. Can you ssh to your eBusd server and try running ebusctl read to check if the value has been broadcasted by the boiler? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread. So I think is not an home-assistant issue too cause ebusctl open a socket connection as my component do — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

I think home-assistant components runs async. Anyway this is not the place to discuss about it, so please only update if you have suggestion or issues with this component.

dkuschmierz commented:
I set the MINIMUM_TIME... to 5 seconds and for example for the CompPressLow i get the following timestamps in the log:

```
2018-10-16 14:01:25 DEBUG (Thread-6) [custom_components.ebus.ebusd] Opening socket to ebusd CompPressLow: read -m 0 -c ehp -f CompPressLow
2018-10-16 14:06:00 DEBUG (Thread-20) [custom_components.ebus.ebusd] Opening socket to ebusd CompPressLow: read -m 0 -c ehp -f CompPressLow
2018-10-16 14:06:55 DEBUG (Thread-21) [custom_components.ebus.ebusd] Opening socket to ebusd CompPressLow: read -m 0 -c ehp -f CompPressLow
2018-10-16 14:08:34 DEBUG (Thread-14) [custom_components.ebus.ebusd] Opening socket to ebusd CompPressLow: read -m 0 -c ehp -f CompPressLow
```

As you can see, the time between the queries is very random. Sorry for discussing this topic here, but whereelse can we do it?

MartinHjelmare commented on homeassistant/components/ebus/ebusd.py:
```diff
@@ -0,0 +1,189 @@
+            _LOGGER.error()
+
+
+class Ebusd(Entity):
```
Why are we adding a sensor entity to a ebus component platform? Shouldn't this be a sensor component platform?

Dinth replied:
I believe that the author tried to pull this as a sensor before, but @balloob has rejected that PR asking CrazYoshi to rewrite this as a component. 
Also there's a huge potential for this component to grow in the future, right now it's mostly reading sensor values, but in the future it may also get binary sensors, switches, or even thermostat functionality.

CrazYoshi replied:
Yes exactly, if you scroll up you can see the change requested by him. That's the reason why I start a new component. Ebus is a generic container where also other component using the same standard can be put inside.

CrazYoshi replied:
So what should I do to this component again? I didn't get
Can you give me some suggestions?

balloob replied:
Component manages connection and hosts services, sensor platform of ebusd will show some info as a sensor entity.

dkuschmierz commented:
No, as there are parameters which can also be written...

> Am 20.10.2018 um 14:36 schrieb Martin Hjelmare <notifications@github.com>:
> 
> @MartinHjelmare commented on this pull request.
> 
> In homeassistant/components/ebus/ebusd.py:
> 
> > +        name = call.data.get('name')
> +        value = call.data.get('value')
> +        command = WRITE_COMMAND.format(self._circuit, name, value)
> +
> +        try:
> +            _LOGGER.debug("Opening socket to ebusd %s: %s", name, command)
> +            command_result = ebusdpy.send_command(self._address, command)
> +            if 'done' not in command_result:
> +                _LOGGER.warning('Write command failed: %s', name)
> +        except socket.timeout:
> +            _LOGGER.error("socket timeout error")
> +        except socket.error:
> +            _LOGGER.error()
> +
> +
> +class Ebusd(Entity):
> Why are we adding a sensor entity to a ebus component platform? Shouldn't this be a sensor component platform?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

CrazYoshi committed:
add automatic state

add automatic state

zyzakin commented on homeassistant/components/ebus/ebusd.py:
```diff
@@ -0,0 +1,189 @@
+            elif self._type == 1:
+                self._state = timer_format(self.data.value[self._name])
+            elif self._type == 2:
+                if self.data.value[self._name] == 1:
```
I would also add the recognition of such a value as 'on' ` or self.data.value[self._name] == 'on' ` , , because frequent are just such answers

CrazYoshi replied:
there is the STATE_ON for that. In this way it will use also translation

dkuschmierz commented:
Hi @CrazYoshi 

would it be possible to move from MIN_TIME_INTERVALL to SCAN_INTERVAL, so we would be able to define different ebusd platforms with different SCAN_INTERVALS. I tried to implement this, but didn't get it work...

balloob commented:
This PR seems to have gone stale. Closing it.
